File: RPC10-18/pells.py
serie = []
f1 = 0
f2 = 1
serie.append(f1)
serie.append(f2)
for i in range(50000):
    f1, f2 = f2, (2*f2)+f1
    serie.append(f2)


while True:
    try:
        a = input()
        ban = False
        if(a == '\n'):
            break
        a = int(a)
        if (serie[a] % 2 == 0):
            print("Pell's number", a, "is divisible by", 2)
            ban = True
        if (serie[a] % 3 == 0):
            print("Pell's number", a, "is divisible by", 3)
            ban = True
        if (serie[a] % 4 == 0):
            print("Pell's number", a, "is divisible by", 4)
            ban = True
        if (serie[a] % 7 == 0):
            print("Pell's number", a, "is divisible by", 7)
            ban = True
        if (serie[a] % 13 == 0):
            print("Pell's number", a, "is divisible by", 13)
            ban = True
        if ban == False:
            print(-1)

    except:
        break

File: RPC10-18/pells.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;


public class pells{

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
		String s;
	
		BigInteger b[] = new BigInteger[50001];
		b[0]=BigInteger.ZERO;
		b[1]=BigInteger.ONE;
		b[2]=BigInteger.valueOf(2);
		int v2[] = {2,3,4,7,13};
		
		int c = 0;
		for (int i = 3; i < b.length; i++) {
			b[i]=b[i-2].add(BigInteger.valueOf(2).multiply(b[i-1]));
		}
		//System.out.println(b[50000]);
		
		while ((s=in.readLine())!=null) {
			String v[]=s.split(" ");
			int a = Integer.parseInt(v[0]);
			
		
			boolean r = false;
			for (int i = 0; i < v2.length; i++) {
				if (b[a].mod(BigInteger.valueOf(v2[i])).compareTo(BigInteger.ZERO)==0) {
					System.out.println("Pell's number " + a + " is divisible by " + v2[i]);
					r = true;
				}
			}
			if (!r) {
				System.out.println("-1");
			}
		}
	}

}
